### ISSUE TITLE
[origin] Collect dc from all namespaces

### ISSUE THREAD

**Issue by wushiqinlou** [sosreport/sos#1410]
Red Hat insights AIR tool is used to detect any issue exists with checking the content of sosreport. For OpenShift part, it is necessary to check all of pods, services, pv, pvc, dc... However, origin.py only collects dc from default namespace currently, and the comments say that "For now just leaving this comment here until we decide if it's worth collecting:". Is it OK to collect dc from all namespaces now?

**Comment by TurboTurtle** [comment]
So AFAIK, the pods, services, pvs, pvcs etc... should all be handled by the `kubernetes` plugin, which today supports collecting from all namespaces, either separately for in one pass (via the `kubernetes.all` option). 

The dc stuff is _also_ retrievable from kubernetes directly - it's not an origin specific resource, so I'm thinking that we should instead remove it's collection from the `origin` plugin and just add it to the resource list in the `kubernetes` plugin since all the namespace handling is already.... handled there. 

Would that work for AIR? 

@codificat thoughts on moving deploymentconfig collection out of origin and into the base kubernetes plugin?

This would probably be a good time to review all the kube resources we collect anyways and see if we need to expand that list.

**Comment by wushiqinlou** [comment]
@TurboTurtle - Thanks for your info. I agree on your comment. In AIR tool, there is code to parse the kubectl_get files from kubernetes directory. Moving dc collection from origin to kubernetes is much pretty good for AIR tool.

**Comment by TurboTurtle** [comment]
Ah, ok - looks like DCs have already been deprecated in kubernetes:

```
# rhel 7.5, latest available kube is 1.5 since straight kubernetes was deprecated on rhel
[root@kub1 ~]# rpm -q kubernetes
kubernetes-1.5.2-0.7.git269f928.el7.x86_64
[root@kub1 ~]# kubectl get --help | grep dc
  * buildconfigs (aka 'bc')  
  * deploymentconfigs (aka 'dc')  
```

vs.

```
# kube 1.10 on Fedora
[jake@terra sos]$ rpm -q kubernetes
kubernetes-1.10.1-0.fc28.x86_64
[jake@terra sos]$ kubectl get --help | grep deploy
  * deployments (aka 'deploy')  
```


As for namespace restriction - I think having a static set of namespaces to collect from would be fine for a default collection - but we could add a plugin option to grab everything if really needed.

As for the `kubernetes` plugin - we can add `deployments` there. Are there any other additional resources we should grab? daemonsets? roles? ingresses? others?

**Comment by wushiqinlou** [comment]
@codificat @TurboTurtle - Thanks for your info. For namespace restriction, I agree on @TurboTurtle's comment. Currently, no other additional resources we should grab.
